imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -392,8 +555,21 @@ public void processCsrs(File file) {
                         saveCount++;
                     }
 
-                    processCsrEvent(csr.toDto(), distributedProcessing);
-                    processCount++;
+                    if (chunkSize > 1) {
+                        chunk.add(csr.toDto());
+                        if (chunk.size() >= chunkSize || lineNumber >= lineCount) {
+                            LOGGER.info(String.format("Distributing chunk %d of %d (%d csrs): %s", chunkNumber,
+                                    chunkCount, chunk.size(), chunkTimer.frequency(chunkNumber)));
+                            String chunkName = String.format("Chunk%d/%d", chunkNumber, chunkCount);
+                            distributeChunk(fileName, chunkName, chunk);
```

Comment:
distributeChunk can throw a chunking exception but no special catch for it here?